File: backend/app/schemas/compiler.py
from datetime import datetime
from pydantic import BaseModel


class CompileRequest(BaseModel):
    node_id: str
    source_code: str
    compile_flags: list[str] = ["-O2", "-march=native"]
    extra_libs: list[str] = []


class CompileResponse(BaseModel):
    status: str           # ok | error
    so_hash: str | None = None
    stderr_output: str | None = None
    compiled_at: datetime | None = None

File: frontend/src/api/compiler.ts
import client from './client'

export interface CompileRequest {
  node_id: string
  source_code: string
  compile_flags?: string[]
  extra_libs?: string[]
}

export interface CompileResult {
  status: 'ok' | 'error'
  so_hash?: string
  stderr_output?: string
  compiled_at?: string
}

export const compilerApi = {
  compile: (req: CompileRequest): Promise<CompileResult> =>
    client.post('/compile', req).then((r) => r.data),

  status: (hash: string): Promise<CompileResult> =>
    client.get(`/compile/${hash}/status`).then((r) => r.data),

  downloadSdk: (): string => '/api/compile/sdk',   /* Direct download URL */
}
